api/commands/compile.ts

```diff
@@ -1,40 +1,68 @@
 import * as protos from "@dataform/protos";
 
-import { fork } from "child_process";
+import { ChildProcess, fork } from "child_process";
 import * as fs from "fs";
 import { promisify } from "util";
 import * as path from "path";
+import { ICompileIPCParameters, ICompileIPCResult } from "../vm/compile";
 
-export function compile(projectDir: string): Promise<protos.CompiledGraph> {
+export async function compile(
+  projectDir: string,
+  defaultSchemaOverride?: string,
```

Comment:
It's inevitably a short matter of time until we need to add more things here. Would be tempted to but them into some CompileOptions interface and pass that as the second param. This is going to have to be serialized imminently too, might want a proto type to hold these